File: JavaApp/src/main/java/com/charlezz/javaapp/di/AppComponent.java
package com.charlezz.javaapp.di;

import com.charlezz.javaapp.App;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {AndroidSupportInjectionModule.class, ActivityModules.class, AppModule.class})
public interface AppComponent extends AndroidInjector<App> {
    @Component.Factory
    abstract class Factory implements AndroidInjector.Factory<App> {

    }
}

File: app/src/main/java/com/charlezz/finalarchitecture/di/AppComponent.kt
package com.charlezz.finalarchitecture.di

import com.charlezz.finalarchitecture.App
import com.charlezz.finalarchitecture.di.module.ActivityModule
import com.charlezz.finalarchitecture.di.module.AppModule
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [(AndroidSupportInjectionModule::class), (ActivityModule::class), (AppModule::class)])
interface AppComponent : AndroidInjector<App> {
    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<App>()
}